typescript/src/lib/structures/SlashCommandStore.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * This is the most basic of examples for a store, this is all you really need
+ * to actively register a store with Sapphire, and anything in the supplied
+ * folder will load if it extends a Sapphire piece.
+ */
+import { SlashCommand } from './SlashCommandPiece';
+import { Store } from '@sapphire/framework';
+
+export class SlashCommandStore extends Store<SlashCommand> {
+	constructor() {
+		// This is the name of the directory we want to look in for our slash
+		// commands.
+		super(SlashCommand, { name: 'slashCommands' });
+	}
+
+	async registerCommands() {
+		const client = this.container.client;
+		if (!client) return;
+
+		// This will split the slash commands between global and guild only.
+		const slashCommands = this.container.stores.get('slashCommands');
+		// @ts-expect-error It is not undefined
```

Comment:
if it's not undefined, you don't need `slashCommands?` < question mark there.
That question mark indicates that it could possible be undefined. If it's undefined, you can't deconstruct an array.

Reply:
use of `@ts-expect-error` is very bad practice if it can be resolved easily

Reply:
I tried the ? But it the error didn't go away so I had to

Reply:
`slashCommands!.`

Reply:
nah either didnt remove the error, lol

Reply:
maybe it was my tsserver problem

Reply:
Do you get the error with some weird "ts server" or also with just compiling with `tsc`?